contracts/GameJam.sol

```diff
@@ -60,19 +93,20 @@ contract GameJam {
   // finish function is a payable used to declare the winner via their address
   // the function should only be called able when the GameJam is in Registration
   function start() public payable
-    onlyAtStage(Stages.Registration)
+    onlyGameJamAdmin //Only an admin can start the GameJam
+    onlyAtStage(Stages.Registration) //GameJam can only be start when in Registration State
     transitionNext {
-
       emit GameJameStarted(now);
     }
 
   // finish function is a payable used to declare the winner via their address
   // the function should only be called able when the GameJam is in Progress
   function finish(address winner) public payable
+    onlyGameJamAdmin //Only an admin can finish the GameJam
     onlyAtStage(Stages.InProgress)
     transitionNext {
-
-
+      //Ensure the winner is a registered competitor. The mapping result is initialised 0x0 so if winner is not in competitors then 0x0 is returned
+      require(keccak256(bytes(competitors[winner])) != keccak256(bytes("")), "Winner should be a registered competitor");
```

Comment:
This check could put the contract into a unfinishable state if there is a possibility for : 

- No competitors registered 
- A competitor registered without a valid IPFS hash

Reply:
Hmmm, yeah you're right, should we add checks into stop this from happening?

I'm happy to make a task for this in trello, and worry about that later, to get this merged in?

wdyt?

Reply:
I think there are a few checks already in place, wondering do you see any holes here: 

- When a competitor is added we do a check to ensure they are entering a valid length ipfs hash

When checking if the winner is a registered competitor, we are checking for two things in my research;
 if an address is not in the `competitors` mapping, `competitors[winner]` will return the null value for that type. If the check passes it confirms:

- `winner ` is in the `competitors[]` mapping
- `competitors[winner]` has a value 

Is there any issue with the above or do you think I'm missing some sort of check?

Reply:
While Im at it refactored the require to something I seen you suggest in previous MR.

Reply:
> When a competitor is added we do a check to ensure they are entering a valid length ipfs hash

For this, I'm not 100% sure if we want to use a IPFS hash or a less strict file location e.g `api.gamejam.io/game-location`, with the ability to change the location. So for the moment it's perfect with one check. We may change this soon.

> When checking if the winner is a registered competitor, we are checking for two things in my research;
if an address is not in the competitors mapping, competitors[winner] will return the null value for that type

I think that's perfect! Maybe something will pop up when we start on the [delegate proxy](https://fravoll.github.io/solidity-patterns/proxy_delegate.html) or the e2e tests that we didn't think of rn. But we can cross that bridge when we come to it.

:gem: